File: unit4/Q6.java
/*Create a class telephone with ( ) , lift ( ) and disconnected ( ) methods as abstract methods 
create another class smart telephone and demonstrate polymorphism*/

abstract class Telephone {
  abstract void call();
  abstract void lift();
  abstract void disconnected();
}

class SmartTelephone extends Telephone {
  void call() {
    System.out.println("call function implemented");
  }
  void lift() {
    System.out.println("lift function implemented");
  }
  void disconnected() {
    System.out.println("disconnected function implemented");
  }
}

class Q6 {
  public static void main (String args[]) {
    Telephone t = new SmartTelephone();
    t.call();
    t.lift();
    t.disconnected();
  }
}

File: unit4/python codes/Q6.py
# Create a class telephone with ( ) , lift ( ) and disconnected ( ) methods as abstract methods 
# create another class smart telephone and demonstrate polymorphism

import abc  #abstract base class

class Telephone(metaclass=abc.ABCMeta):
  @abc.abstractmethod
  def call(self):
    pass

  @abc.abstractmethod
  def list(self):
    pass

  @abc.abstractmethod
  def disconnected(self):
    pass

class SmartTelephone(Telephone):
  def call(self):
    print("Call function called")

  def lift(self):
    print("Lift function called")

  def disconnected(self):
    print("Disconnected function called")

